src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -60,14 +59,21 @@ bool bypassBrowserSecurity
 			return new( Org: org, User: user, Client: oktaAuthenticated );
 		}
 
-		oktaAuthenticated = await GetDssoAuthenticatedClientAsync(
-			orgUrl,
-			user,
-			nonInteractive,
-			bypassBrowserSecurity
-		);
-		if( oktaAuthenticated is not null ) {
-			return new( Org: org, User: user, Client: oktaAuthenticated );
+		if( TryGetDssoAuthOptions( nonInteractive, bypassBrowserSecurity, out var dssoAuthOptions ) ) {
+			if( !nonInteractive ) {
+				Console.Error.WriteLine( "Attempting Okta passwordless authentication..." );
```

Comment:
`TryGetDssoAuthOptions` contains early validation for DSSO/browser process. It's split out mainly to better control when we print the "attempting..." and "falling back to..." messages.

Some points in consideration:

* When we do launch a browser, it takes a couple seconds, so the "attempting passwordless..." message is worthwhile to print
* If DSSO fails after a browser is launched and the "attempting" message is printed, it makes sense to print another message indicating that we're now switching to password auth
* It only makes sense to print the "fallback" message if the "attempting" message is printed
* It only makes sense to print the "attempting" message if we're actually launching a browser, because deciding whether to launch the browser is very fast (synchronous code)
* Therefore, it makes sense to control when we print both of these messages in the same code area